src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,57 @@
+package seedu.duke;
+import storage.Storage;
+import storage.TaskList;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+public class Duke {
+
+    private final Ui ui;
+    private final Parser parser;
+    private final Storage storage;
+    private final TaskList tasks;
+
+    private Duke() throws IOException, ParseException {
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage();
+        tasks = new TaskList(storage.read());
+    }
+
+    private void run() {
+
+        ui.showWelcome();
+        boolean isQuit = false;
+
+        while (!isQuit) {
+            try {
+
+                String input = ui.getCommand();
+                String[] cmd = parser.splitCmd(input);
+
+                if (parser.checkQuit(cmd[0].trim())) {
+                    isQuit = true;
+                } else {
+                    String task = cmd.length < 2 ? "" : cmd[1];
+                    parser.runCmd(cmd[0].trim(), task.trim(), tasks, ui, storage);
+                }
+
+            } catch (DukeException | IOException e) {
+                ui.showError(e.toString());
+            }
```

Comment:
```suggestion
            try {
                String input = ui.getCommand();
                String[] cmd = parser.splitCmd(input);

                if (parser.checkQuit(cmd[0].trim())) {
                    isQuit = true;
                } else {
                    String task = cmd.length < 2 ? "" : cmd[1];
                    parser.runCmd(cmd[0].trim(), task.trim(), tasks, ui, storage);
                }
            } catch (DukeException | IOException e) {
                ui.showError(e.toString());
            }
```
Good line breaks to make it the logic clear and readable!
But I do think that there is a need for the line breaks after try and before catch